src/badge/WithBadge.js

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { Platform, StyleSheet, View } from "react-native";
+import { Badge } from "react-native-elements";
+
+const styles = StyleSheet.create({
+  badge: {
+    backgroundColor: Platform.select({
+      android: "#f44336",
+      ios: "#ff3b30"
+    }),
+    borderRadius: 9,
+    height: 18,
+    shadowColor: "black",
+    shadowOffset: {
+      width: 0,
+      height: 0.5
+    },
+    shadowRadius: 0.5,
+    shadowOpacity: 1.0,
+    padding: 0,
+    width: 18
+  },
+  badgeContainer: {
+    position: "absolute"
+  },
+  badgeText: {
+    fontSize: 10
+  }
+});
+
+const withBadge = (
+  value,
+  offSetX = 10,
+  offSetY = 22,
+  badgeProps = {},
+  hidden = typeof value === "number" ? value === 0 : value !== ""
+) => WrappedComponent =>
+  class extends React.Component {
+    render() {
+      const badgeValue = typeof value === "function" ? value(this.props) : value;
+      return (
+        <View>
+          <WrappedComponent {...this.props} />
+          {!hidden && (
+            <Badge
+              containerStyle={styles.badge}
+              textStyle={styles.badgeText}
+              value={badgeValue}
+              wrapperStyle={[styles.badgeContainer, { bottom: offSetY, left: offSetX }]}
```

Comment:
Also you should probably look over the Badge. We've made some changes to badge on next as apart of the 1.0.0 release. So you might want to check those changes first. 

Here's the PR https://github.com/react-native-training/react-native-elements/pull/1545

_PS_ there's no more `wrapperStyle`

Reply:
Thank you, I edited the commit. If you are okay with the code and the API, I would like to also add the docs etc. to this PR 😊